parsl/executors/high_throughput/process_worker_pool.py

```diff
@@ -149,6 +174,7 @@ def pull_tasks(self, kill_event):
                         #    [i['task_id'] for i in self.pending_task_queue]))
             else:
                 logger.debug("[TASK_PULL_THREAD] No incoming tasks")
+                poll_timer = poll_timer * 2
```

Comment:
if i have a workflow that has an initial stage that takes 1h and doesn't do much, and then generates 1000 smaller tasks, and i allocated 1000 nodes, does this mean that 999 workers might wait "a long time" (~1h?) after the 1000 smaller tasks become ready before they begin work, due to poll_timer getting very big over that initial hour?

Reply:
Good catch. I've added a condition to limit timeout increase to self.heartbeat_period